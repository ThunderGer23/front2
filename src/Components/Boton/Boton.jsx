import PropTypes from 'prop-types'

export const Boton = ({texto}) => {
  return(
    <>
      <div role='div' className="d-grid gap-2 mb-4">
          <button name={texto+'button'} className="btn btn-light" type="button">
              <i role='i' className="fa fa-right-to-bracket orange"></i>
              {texto}
          </button>
      </div>
    </>
  )
}

Boton.protoTypes = {
  texto: PropTypes.string.isRequired
}