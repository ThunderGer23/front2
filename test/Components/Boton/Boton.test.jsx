import {render,screen} from "@testing-library/react"
import { Boton } from "../../../src/Components/Boton/Boton"

describe('Pruebas en el Boton',()=>{
    const texto= 'Boton'

    test('Esta prueba es para hacer match con el snapshot',() =>{
        const {container} = render(<Boton texto={texto}/>)//Renderiza el componente
        expect(container).toMatchSnapshot();//Toma una "foto"
    })

    test('Esta prueba valida que el div exista y pertenezca a la clase correcta',() =>{
        render(<Boton texto={texto}/>)
        //expect(screen.getAllByRole('div')).toBeTruthy()//Se valida que exista el div
        const {container} = screen.getAllByRole('div')
        const {className}= screen.getByRole('div')//Se salva la clase del div del componente en className
        console.log(container)
        expect(className).toBe('d-grid gap-2 mb-4')//Se valida que el div tenga la misma clase que className
    })

    
})