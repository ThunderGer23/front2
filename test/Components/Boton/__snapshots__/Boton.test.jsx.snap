// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Boton Esta prueba es para hacer match con el snapshot 1`] = `
<div>
  <div
    class="d-grid gap-2 mb-4"
  >
    <button
      class="btn btn-light"
      type="button"
    >
      <i
        class="fa fa-right-to-bracket orange"
      />
      Boton
    </button>
  </div>
</div>
`;
